tests/yam/agama/agama.pm

```diff
@@ -52,12 +53,16 @@ sub run {
     return if get_var('INST_ABORT');
 
     # make sure we will boot from hard disk next time
-    if (is_s390x && is_svirt) {
+    if (is_s390x() && is_svirt()) {
         select_console 'installation';
         my $svirt = console('svirt')->change_domain_element(os => boot => {dev => 'hd'});
     }
 
-    $reboot_page->reboot();
+    (is_s390x() || is_ppc64le()) ?
+      # reboot via console
+      power_action('reboot', keepconsole => 1, first_reboot => 1) :
+      # graphical reboot
+      $reboot_page->reboot();
```

Comment:
Does it mean that it is rebooted twice? First reboot vis console and second in gfx ui?

Reply:
if (is_s390x() || is_ppc64le()) {
    # reboot via console
    power_action('reboot', keepconsole => 1, first_reboot => 1);
} else {
    # graphical reboot
    $reboot_page->reboot();
}

If it is written in this way, it will be clearer, of course, without affecting anything.

Reply:
no, it is a ternary,  you have a condition at the beginning and then you have lines indented below with each option, it is really useful for having each line commented with something special you do, I prefer those than the typical `{ }` for this cases, it is less verbose, specially when you don't plan to have more complex actions in each line.
```
condition ?
  branch1 :
  branch2 :
```

Reply:
Nice, i'm used mainly to ternary one-liners and my brain  read `:` as `;` at the end of line :)